apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeEditor.tsx

```diff
@@ -410,6 +410,45 @@ export const CodeEditor = ({ sourcesFolderId, setValue }: Props) => {
     [dispatch, sourcesFolderId],
   );
 
+  const handleToggleFolder = useCallback(
+    (folderId: string) => {
+      if (openedFoldersIds.includes(folderId)) {
+        const childFoldersIds = getChildAndCurrentFoldersIdsById(
+          folderId,
+          folders,
+        );
+        setOpenedFoldersIds(
+          openedFoldersIds.filter((id) => !childFoldersIds.includes(id)),
+        );
+      } else {
+        setOpenedFoldersIds(openedFoldersIds.concat(folderId));
+        const folder = folders.find((f) => f.id === folderId);
+        if (folder?.status !== UploadStatus.LOADED) {
+          dispatch(FilesActions.getFilesWithFolders({ id: folderId }));
+        } else {
+          setOpenedFoldersIds(openedFoldersIds.concat(folderId));
+          const folder = folders.find((f) => f.id === folderId);
+          if (folder?.status !== UploadStatus.LOADED) {
+            dispatch(
+              FilesActions.getFilesWithFolders({
+                id: folderId,
+              }),
+            );
+          }
+        }
+      }
+    },
+    [openedFoldersIds, setOpenedFoldersIds, folders, dispatch],
+  );
+
+  const handleAddFolder = useCallback(
+    (folderId: string) => {
+      dispatch(FilesActions.addNewFolder({ parentId: folderId }));
+      handleToggleFolder(folderId);
+    },
+    [openedFoldersIds, dispatch, handleToggleFolder],
```

Comment:
React Hook useCallback has an unnecessary dependency: 'openedFoldersIds'. Either exclude it or remove the dependency array